code/datums/components/electrified_buckle.dm

```diff
@@ -0,0 +1,184 @@
+#define ELECTRIC_BUCKLE_WAIT_TIME 5 SECONDS
+///divide the power in the cable net under parent by this to determine the shock damage
+#define ELECTRIC_BUCKLE_SHOCK_STRENGTH_DIVISOR  3000
+///it will not shock the mob buckled to parent if its required to use a cable to shock and the cable has less than this power availaible
+#define ELECTRIC_BUCKLE_MINUMUM_POWERNET_STRENGTH 10
+
+/datum/component/electrified_buckle
+	///if usage_flags has SHOCK_REQUIREMENT_ITEM, this is the item required to be inside parent in order for it to shock buckled mobs
+	var/obj/item/required_object
+	///this is casted to the overlay we put on parent_chair
+	var/list/requested_overlays
+	///it will only shock once every ELECTRIC_BUCKLE_WAIT_TIME
+	COOLDOWN_DECLARE(electric_buckle_cooldown)
+	///these flags tells this instance what is required in order to allow shocking
+	var/usage_flags
+	///if true, this will shock the buckled mob every ELECTRIC_BUCKLE_WAIT_TIME in process()
+	var/shock_on_loop = TRUE
+	///how much damage is done per shock iff usage_flags doesnt have SHOCK_REQUIREMENT_LIVE_CABLE
+	var/shock_damage = 50
+	///this signal was given as an argument to register for parent to emit, if its emitted to parent then shock_on_demand is called. var is so it can be unregistered
+	var/requested_signal_parent_emits = null
+
+/datum/component/electrified_buckle/Initialize(input_requirements, obj/item/input_item, list/overlays_to_add, override_buckle, damage_on_shock = 50, signal_to_register_from_parent)
+	var/atom/movable/parent_as_movable = parent
+	if(!istype(parent_as_movable))
+		return COMPONENT_INCOMPATIBLE
+
+	usage_flags = input_requirements
+
+	if(!parent_as_movable.can_buckle && !override_buckle)
+		return COMPONENT_INCOMPATIBLE
+	else if (override_buckle)
+		parent_as_movable.can_buckle = TRUE
+
+	if((usage_flags & SHOCK_REQUIREMENT_ITEM) && QDELETED(input_item))
+		return COMPONENT_INCOMPATIBLE
+
+	if(HAS_TRAIT(parent_as_movable, TRAIT_ELECTRIFIED_BUCKLE))
+		return COMPONENT_INCOMPATIBLE
+
+	if(usage_flags & SHOCK_REQUIREMENT_ITEM)
+		required_object = input_item
+		required_object.Move(parent_as_movable)
+		RegisterSignal(required_object, COMSIG_PARENT_PREQDELETED, .proc/delete_self)
+		RegisterSignal(parent, COMSIG_ATOM_TOOL_ACT(TOOL_SCREWDRIVER), .proc/move_required_object_from_contents)
+
+		if(usage_flags & SHOCK_REQUIREMENT_ON_SIGNAL_RECEIVED)
+			shock_on_loop = FALSE
+			RegisterSignal(required_object, COMSIG_ASSEMBLY_PULSED, .proc/shock_on_demand)
+		else if(usage_flags & SHOCK_REQUIREMENT_SIGNAL_RECEIVED_TOGGLE)
```

Comment:
This is a bitfield, is it not possible to have both?

Reply:
you can give it both (currently nothing uses ON_SIGNAL_RECEIVED though, i just assumed someone in the future might want it) but they do different things and are activated by the same method. ON_SIGNAL_RECEIVED disables the shocking loop entirely and only shocks when you pulse required_object (the code assumes its an assembly of some kind that can be pulsed). SIGNAL_RECEIVED_TOGGLE just toggles the shocking loop when you do the same. 

if you had both then it would toggle the shocking loop and also shock manually whenever you did it which doesn't really have a use since the shock loop is only 5 seconds